examples/data-objects/todo/src/Todo/TodoListView.tsx

```diff
@@ -0,0 +1,122 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { CollaborativeInput } from "@fluid-example/example-utils";
+import { SharedString, type ISharedString } from "@fluidframework/sequence/legacy";
+import { Tree, type TreeNode } from "@fluidframework/tree/legacy";
+import React, { useEffect, useRef, useState } from "react";
+
+import { TodoItemView } from "../TodoItem/index.js";
+
+// eslint-disable-next-line import/no-unassigned-import
+import "./style.css";
+
+import { type TodoListDataObject } from "./index.js";
+
+export interface TodoViewProps {
+	readonly todoModel: TodoListDataObject;
+	readonly getDirectLink: (itemId: string) => string;
+}
+
+// TODO: This was copied over from the "@fluid-experimental/tree-react-api" package.
+// This should be imported from that package, once it is no longer experimental.
+export function useTree(subtreeRoot: TreeNode): void {
+	// Use a React effect hook to invalidate this component when the subtreeRoot changes.
+	// We do this by incrementing a counter, which is passed as a dependency to the effect hook.
+	const [invalidations, setInvalidations] = useState(0);
+
+	// React effect hook that increments the 'invalidation' counter whenever subtreeRoot or any of its children change.
+	useEffect(() => {
+		// Returns the cleanup function to be invoked when the component unmounts.
+		return Tree.on(subtreeRoot, "treeChanged", () => {
+			setInvalidations((i) => i + 1);
+		});
+	}, [invalidations, subtreeRoot]);
+}
+
+export const TodoView: React.FC<TodoViewProps> = (props: TodoViewProps) => {
+	const { todoModel, getDirectLink } = props;
+	const [titleString, setTitleString] = useState<SharedString | undefined>();
+
+	const newItemTextInputRef = useRef<HTMLInputElement>(null);
+
+	useTree(todoModel.treeView.root);
+
+	useEffect(() => {
+		Promise.resolve(todoModel.treeView.root.title.get())
+			.then((title) => {
+				setTitleString(title as ISharedString);
+			})
+			.catch((error) => {
+				console.log("todomodel");
+				console.log(todoModel);
+				console.error(error);
+			});
+		return () => {};
+	}, [todoModel]);
+
+	if (titleString === undefined) {
+		return <div>Loading...</div>;
+	}
+	const handleCreateClick = async (ev: React.FormEvent<HTMLFormElement>): Promise<void> => {
+		ev.preventDefault();
+		if (newItemTextInputRef.current === null) {
+			throw new Error("New item text field missing");
+		}
+		await todoModel.addTodoItem({
+			startingText: newItemTextInputRef.current.value,
+		});
+		newItemTextInputRef.current.value = "";
+	};
```

Comment:
I would recommend making this non-async. E.g.
```suggestion
	const handleCreateClick = (ev: React.FormEvent<HTMLFormElement>): void => {
		ev.preventDefault();
		if (newItemTextInputRef.current === null) {
			throw new Error("New item text field missing");
		}
		todoModel.addTodoItem({
			startingText: newItemTextInputRef.current.value,
		}).then(() => {
			newItemTextInputRef.current.value = "";
		});
	};
```

Then you can remove the linter rule disables lower in the function

Reply:
Updated to be non-async, and removed unused lint suppression